diff --git a/hie-plugin-api/Haskell/Ide/Engine/GhcUtils.hs b/hie-plugin-api/Haskell/Ide/Engine/GhcUtils.hs
index fbe5a1e6..bc48f98c 100644
--- a/Haskell/Ide/Engine/GhcUtils.hs
+++ b/Haskell/Ide/Engine/GhcUtils.hs
@@ -17,7 +17,7 @@ import           Haskell.Ide.Engine.PluginUtils (ErrorHandler(..))
 -- Convert progress continuation to a messager
 toMessager :: (Core.Progress -> IO ()) -> G.Messager
 toMessager k _hsc_env (nk, n) _rc_reason ms =
-  let prog = Core.Progress (Just (fromIntegral nk/ fromIntegral n))  (Just mod_name)
+  let prog = Core.Progress (Just (100 * fromIntegral nk / fromIntegral n))  (Just mod_name)
       mod_name = T.pack $ moduleNameString (moduleName (ms_mod ms))
   in k prog
 
